pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -558,6 +574,12 @@ func (ssr *Snapshotter) snapshotEventHandler(stopCh <-chan struct{}) error {
 			if err != nil {
 				return err
 			}
+			if err = heartbeat.UpdateFullSnapshotLeases(ssr.logger, ssr.PrevFullSnapshot, ssr.k8sClientset); err != nil {
```

Comment:
Please add a check like `if b.config.HealthConfig.Enabled` so that it doesn't attempt to renew the leases unless lease renewal is enabled.